client/src/boot/registerReducers.js

```diff
@@ -0,0 +1,33 @@
+import Injector from 'lib/Injector';
+import { combineReducers } from 'redux';
+import { reducer as ReduxFormReducer } from 'redux-form';
+import { routerReducer } from 'react-router-redux';
+import ConfigReducer from 'state/config/ConfigReducer';
+import SchemaReducer from 'state/schema/SchemaReducer';
+import RecordsReducer from 'state/records/RecordsReducer';
+import BreadcrumbsReducer from 'state/breadcrumbs/BreadcrumbsReducer';
+import TreeDropdownFieldReducer from 'state/treeDropdownField/TreeDropdownFieldReducer';
+// import UnsavedFormsReducer from 'state/unsavedForms/UnsavedFormsReducer';
+import applyFormMiddleware from 'lib/dependency-injection/applyFormMiddleware';
+
+export default (extra = {}) => {
+  const FormReducer = applyFormMiddleware(
+    combineReducers({
+      formState: ReduxFormReducer,
+      formSchemas: SchemaReducer,
+    })
+  );
+
+  Object.keys(extra).forEach((key) => {
+    const reducer = extra[key];
```

Comment:
I think the one `registerReducers` file is fine, the `extra` in the params is mainly for the `apollo.reducer()` which I had no way to access that otherwise.
I don't see any value in splitting out the formReducers from "coreReducers", to me it's just as core